File: src/main/java/me/miki/shindo/management/event/impl/EventCameraRotation.java
package me.miki.shindo.management.event.impl;

import lombok.Getter;
import me.miki.shindo.management.event.Event;

@Getter
public class EventCameraRotation extends Event {

    private float yaw;
    private float pitch;
    private float roll;
    private float thirdPersonDistance;

    public EventCameraRotation(float yaw, float pitch, float roll, float thirdPersonDistance) {
        this.yaw = yaw;
        this.pitch = pitch;
        this.roll = roll;
        this.thirdPersonDistance = thirdPersonDistance;
    }

    public void setYaw(float yaw) {
        this.yaw = yaw;
    }

    public void setPitch(float pitch) {
        this.pitch = pitch;
    }

    public void setRoll(float roll) {
        this.roll = roll;
    }

    public void setThirdPersonDistance(float thirdPersonDistance) {
        this.thirdPersonDistance = thirdPersonDistance;
    }
}

File: src/main/kotlin/me/miki/shindo/management/event/impl/EventCameraRotation.kt
package me.miki.shindo.management.event.impl

import me.miki.shindo.management.event.Event

class EventCameraRotation(
    @JvmField var yaw: Float,
    @JvmField var pitch: Float,
    @JvmField var roll: Float,
    @JvmField var thirdPersonDistance: Float
) : Event() {
    fun getYaw(): Float = yaw
    fun setYaw(yaw: Float) {
        this.yaw = yaw
    }

    fun getPitch(): Float = pitch
    fun setPitch(pitch: Float) {
        this.pitch = pitch
    }

    fun getRoll(): Float = roll
    fun setRoll(roll: Float) {
        this.roll = roll
    }

    fun getThirdPersonDistance(): Float = thirdPersonDistance
    fun setThirdPersonDistance(thirdPersonDistance: Float) {
        this.thirdPersonDistance = thirdPersonDistance
    }
}

